atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,313 @@
+from fireworks import FiretaskBase, FWAction, explicit_serialize
+
+from monty.serialization import loadfn, dumpfn
+
+from atomate.utils.utils import get_logger, env_chk
+from atomate.vasp.database import VaspCalcDb
+
+from datetime import datetime
+import numpy as np
+
+from pymatgen import Structure
+from pymatgen.command_line.bader_caller import bader_analysis_from_path
+from pymatgen.analysis.magnetism.heisenberg import HeisenbergMapper, HeisenbergModel
+from pymatgen.analysis.magnetism import CollinearMagneticStructureAnalyzer, Ordering
+from pymatgen.command_line.vampire_caller import VampireCaller
+
+__author__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+
+logger = get_logger(__name__)
+
+
+@explicit_serialize
+class HeisenbergModelMapping(FiretaskBase):
+    """
+    Map structures and energies to a Heisenberg model and compute exchange
+    parameters for a given NN cutoff.
+
+    * heisenberg_settings: 
+        cutoff (float): Starting point for nearest neighbor search.
+        tol (float): Tolerance for equivalent NN bonds.
+        average (bool): Compute only <J>.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+        structures (list): Magnetic structures.
+        energies (list): Energies / atom (eV).
+
+    Optional parameters:
+        heisenberg_settings (dict): A config dict for Heisenberg model 
+            mapping, detailed above.
+
+    """
+
+    required_params = [
+        "db_file",
```

Comment:
It looks like `db_file` and `wf_uuid` are no longer required btw.

Reply:
Fixed!